src/SFA.DAS.EAS.Web/Orchestrators/EmployerCommitmentsOrchestrator.cs

```diff
@@ -751,16 +743,35 @@ private Task<GetAccountTransferConnectionsResponse> GetTransferConnectionsNoAuth
                     AccountId = accountId
                 });
 
-                var commitmentStatuses = data.Commitments
+                //todo: call into commitments api or db to get counts, seems excessive to fetch all cohorts data just to count
```

Comment:
I can see we will need to do counts here, but is it possible to refactor this code into a single calculation. Are we not just making the code more complicated by extending the current logic? Also as you say, perhaps this logic should be in the API layer and that should return all the summary counts

Reply:
This code has already been refactored in DPP-1229 - https://github.com/SkillsFundingAgency/das-employercommitments/pull/117